pilot/pkg/networking/plugin/plugin.go

```diff
@@ -109,6 +124,12 @@ type Plugin interface {
 	// added to RDS in the outbound path.
 	OnOutboundRouteConfiguration(in *InputParams, routeConfiguration *xdsapi.RouteConfiguration)
 
+	// OnOutboundRoute is called for every HTTP route in route configurations.
+	OnOutboundRoute(in *InputParams,
```

Comment:
fyi, I was anyway planning to change *OutboundRouteConfiguration to *OutboundRoute (with corresponding changes to how it's called), but I can work off this.

Reply:
I think we need both. RouteConfiguration has hosts on top of routes, so we need a callback for the virtual host entity.

Reply:
having both could get messy since it's unclear which callback would be responsible for updating the routes. unless we make it clear that On*RouteConfiguration can only modify at that level (e.g. add routes to RouteConfiguration) but may not modify any existing routes.

Reply:
Right. But I don't see another way. The whole configuration is tree structured, and we need visit nodes in the plugins. There's another problem with inability to inject/reorder listeners/clusters, which I did not attempt to tackle, which also faces the same problem.